src/refdb_fs.c

```diff
@@ -63,6 +66,18 @@ typedef struct refdb_fs_backend {
 	uint32_t direach_flags;
 } refdb_fs_backend;
 
+/*
+ * List of symbolic reference names that are always stored in the
+ * gitdir. This is required such that we are able to correctly
+ * resolve these references via the gitdir instead of the
+ * commondir when working trees are used.
+ */
+static const char *symrefs[] = {
+    "HEAD",
+    "ORIG_HEAD",
+    NULL,
+};
```

Comment:
Is this seems to be missing `MERGE_HEAD` and `FETCH_HEAD`, or are those handled differently? Is there really a list of these or should we perform the `{_,}HEAD` suffix check like we do to determine if we want to allow writing to a particular ref?